c7n/resources/dms.py

```diff
@@ -105,3 +121,114 @@ def process(self, resources):
         client = local_session(self.manager.session_factory).client('dms')
         for arn, r in zip(self.manager.get_arns(resources), resources):
             client.delete_replication_instance(ReplicationInstanceArn=arn)
+
+
+@ReplicationInstance.action_registry.register('tag')
+class InstanceTag(Tag):
+    """
+    Add tag(s) to a replication instance
+
+    :example:
+
+        .. code-block:: yaml
+
+            policies:
+                - name: tag-dms-required
+                  resource: dms-instance
+                  filters:
+                    - "tag:RequireTag": absent
+                  actions:
+                    - type: tag
+                      key: RequiredTag
+                      value: RequiredTagValue
+    """
+    permissions = ('dms:AddTagsToResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('dms')
+        tags_list = []
+        for t in tags:
+            tags_list.append({'Key': t['Key'], 'Value': t['Value']})
```

Comment:
fwiw, you can collapse these three statements into a single list comprehension.

Reply:
thanks for pointing that out... should have realized i could do that... will update accordingly.